src/patternfly/components/Button/button.scss

```
@@ -564,6 +575,11 @@
 
     color: var(--pf-c-button--disabled--Color);
     background-color: var(--pf-c-button--disabled--BackgroundColor);
+
+    & .pf-c-badge {
+      --pf-c-badge--m-unread--Color: var(--pf-c-button--disabled__c-badge--Color);
+      --pf-c-badge--m-unread--BackgroundColor: var(--pf-c-button--disabled__c-badge--BackgroundColor);
```

Looks like you can probably just remove the disabled border here instead of using `:not()` above to create the border.

```suggestion
      --pf-c-badge--m-unread--BackgroundColor: var(--pf-c-button--disabled__c-badge--BackgroundColor);
      --pf-c-button--m-primary__c-badge--BorderWidth: 0;
```